raiden-ts/src/transfers/epics.ts

```diff
@@ -449,24 +521,50 @@ export const transferGenerateAndSignEnvelopeMessageEpic = (
   | typeof transferExpired
   | typeof transferExpireFailed
   | typeof withdrawSendConfirmation
->> =>
-  state$.pipe(
-    publishReplay(1, undefined, state$ => {
-      const withdrawCache = new LruCache<string, Signed<WithdrawConfirmation>>(32);
-      return action$.pipe(
-        filter(isActionOf([transfer, transferUnlock, transferExpire, withdrawReceiveRequest])),
-        concatMap(action =>
-          isActionOf(transfer, action)
-            ? makeAndSignTransfer(state$, action, deps)
-            : isActionOf(transferUnlock, action)
-            ? makeAndSignUnlock(state$, action, deps)
-            : isActionOf(transferExpire, action)
-            ? makeAndSignLockExpired(state$, action, deps)
-            : makeAndSignWithdrawConfirmation(state$, action, deps, withdrawCache),
-        ),
-      );
+>> => {
+  const withdrawCache = new LruCache<string, Signed<WithdrawConfirmation>>(32);
+  const latestState$ = deps.latest$.pipe(pluckDistinct('state'));
+  return action$.pipe(
+    filter(isActionOf([transfer, transferUnlock, transferExpire, withdrawReceiveRequest])),
+    concatMap(action => {
+      switch (action.type) {
+        case getType(transfer): {
+          return makeAndSignTransfer(latestState$, action, deps);
+        }
+        case getType(transferUnlock): {
+          return makeAndSignUnlock(latestState$, action, deps);
+        }
+        case getType(transferExpire): {
+          return makeAndSignLockExpired(latestState$, action, deps);
+        }
+        default: {
+          return makeAndSignWithdrawConfirmation(latestState$, action, deps, withdrawCache);
+        }
+      }
     }),
   );
+};
+
+const transferSignedRetryMessage = (
+  action$: Observable<RaidenAction>,
+  state$: Observable<RaidenState>,
+  action: ActionType<typeof transferSigned>,
+) => {
```

Comment:
Seems to return an observable, would be great to add a descriptive return type. Should also be `transferSignedRetryMessage$` since it returns an observable, right?